File: leetcode/java/0704. Binary Search.java
package leetcode.java;

// Time complexity: O(logN)
// Space complexity: O(1)

class Solution {
    public int search(int[] nums, int target) {
        int left = 0;
        int right = nums.length - 1;

        while (right > left) {
            int mid = (left + right) / 2;
            if (nums[mid] == target) {
                return mid;
            } else if (nums[mid] > target) {
                right = mid - 1;
            } else {
                left = mid + 1;
            }
        }
        return -1;
    }
}

File: leetcode/python/0704. Binary Search.py
# Time complexity: O(logN)
# Space complexity: O(1)


class Solution(object):
    def search(self, nums, target):

        left = 0
        right = len(nums) - 1

        while right >= left:
            mid = (left + right) / 2

            if nums[mid] == target:
                return mid
            elif nums[mid] < target:
                left = mid + 1
            else:
                right = mid - 1
        return -1